### PR TITLE
Add user to target audience list in mailchimp after newsletter subscription.

### PR DESCRIPTION
**Pull request by lauracernau**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
         "drupal/console": "~1.0",
         "drupal/contact_storage": "1.0-beta9",
         "drupal/core": "8.4.0",
+        "drupal/mailchimp": "^1.3",
```
Comment (corneaalina) [inline composer.json]: Please specify a concrete version for this module.

Comment (ChrisHng) [inline composer.json]: Require specific version of module.

File: docroot/modules/custom/dct_newsletter/src/Form/NewsletterSubscriptionForm.php [hunk 2]
```diff
@@ -166,6 +166,9 @@ public function saveSubscription(array &$form, FormStateInterface $form_state) {
 
         $html = render($html);
         $command = new ReplaceCommand('#dct-newsletter-form', $html);
+
+        $this->addMailchimpUser($form_state->getValue('email'), 'Target Audience');
+
```
Comment (ChrisHng) [inline docroot/modules/custom/dct_newsletter/src/Form/NewsletterSubscriptionForm.php]: No need for this line here

File: docroot/config/sync/core.extension.yml [hunk 3]
```diff
@@ -25,6 +25,9 @@ module:
   history: 0
   image: 0
   link: 0
+  mailchimp: 0
+  mailchimp_campaign: 0
```
Comment (corneaalina) [inline docroot/config/sync/core.extension.yml]: Do we need mailchimp_campaigns enabled?

File: docroot/modules/custom/dct_newsletter/src/Form/NewsletterSubscriptionForm.php [hunk 4]
```diff
@@ -166,6 +166,9 @@ public function saveSubscription(array &$form, FormStateInterface $form_state) {
 
         $html = render($html);
         $command = new ReplaceCommand('#dct-newsletter-form', $html);
+
```
Comment (danielpopdan) [inline docroot/modules/custom/dct_newsletter/src/Form/NewsletterSubscriptionForm.php]: Please add a comment here.

File: docroot/modules/custom/dct_newsletter/src/Form/NewsletterSubscriptionForm.php [hunk 5]
```diff
@@ -189,6 +192,39 @@ public function saveSubscription(array &$form, FormStateInterface $form_state) {
...
+   */
+  protected function addMailchimpUser($user_email, $list_name) {
+    $list_id = $this->findListId($list_name);
+    if ($list_id) {
+      mailchimp_subscribe($list_id, $user_email, NULL, [], FALSE, $format = 'html');
+    }
+  }
+
+  /**
+   * Finds the id of the targt list form Mailchimp.
```
Comment (danielpopdan) [inline docroot/modules/custom/dct_newsletter/src/Form/NewsletterSubscriptionForm.php]: Typo in "target" and "from".

File: docroot/modules/custom/dct_newsletter/src/Form/NewsletterSubscriptionForm.php [hunk 6]
```diff
@@ -189,6 +192,39 @@ public function saveSubscription(array &$form, FormStateInterface $form_state) {
...
+   *
+   * @param $list_name
+   *   The name of the list.
+   *
+   * @return null|string
+   *   The id of the list, if exists, null otherwise.
+   */
+  protected function findListId($list_name) {
```
Comment (danielpopdan) [inline docroot/modules/custom/dct_newsletter/src/Form/NewsletterSubscriptionForm.php]: I would move the methods regarding MailChimp in a new service.

File: docroot/config/sync/core.extension.yml (hunk 3 above)
Comment (lauracernau) [reply docroot/config/sync/core.extension.yml]: Yes, because we are going to create campaigns directly in Drupal, in order to send newsletters.

**Pull request merged**